File: VB/XtraReport1.vb
Imports System
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
Imports DevExpress.Data.Filtering.Helpers
Imports System.Data
Imports DevExpress.Data.Filtering
Imports System.Collections.Generic
Imports DevExpress.Data.Browsing
Imports System.Linq

Namespace dxSample
    Partial Public Class XtraReport1
        Inherits DevExpress.XtraReports.UI.XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub
        Private Sub calculatedField2_GetValue(ByVal sender As Object, ByVal e As GetValueEventArgs) Handles calculatedField2.GetValue
            Dim expression As String = calculatedField_Base.Expression
            Dim currentRow As Object = GetCurrentRow()
            Dim dataSource As Object = If(calculatedField_Base.DataSource Is Nothing, DataSource, calculatedField_Base.DataSource)
            Dim dataMember As String = calculatedField_Base.DataMember

            Dim props As PropertyDescriptorCollection
            Using context As New DataContext()
                props = context.GetListItemProperties(dataSource, dataMember)
            End Using
            Dim result As Object = (New ExpressionEvaluator(props, CriteriaOperator.Parse(expression))).Evaluate(currentRow)

            e.Value = result
        End Sub
    End Class
End Namespace

File: CS/XtraReport1.cs
using System;
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using DevExpress.Data.Filtering.Helpers;
using System.Data;
using DevExpress.Data.Filtering;
using System.Collections.Generic;
using DevExpress.Data.Browsing;
using System.Linq;

namespace dxSample {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }
        private void calculatedField2_GetValue(object sender, GetValueEventArgs e) {
            string expression = calculatedField_Base.Expression;
            object currentRow = GetCurrentRow();
            object dataSource = calculatedField_Base.DataSource == null ? DataSource : calculatedField_Base.DataSource;
            string dataMember = calculatedField_Base.DataMember;

            PropertyDescriptorCollection props;
            using(DataContext context = new DataContext()) {
                props = context.GetListItemProperties(dataSource, dataMember);
            }
            object result = new ExpressionEvaluator(props, CriteriaOperator.Parse(expression)).Evaluate(currentRow);
            
            e.Value = result;
        }
    }
}
